Autonomous decision making demonstration.ipynb

```diff
@@ -176,24 +176,21 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 5,
+   "execution_count": 16,
    "metadata": {},
    "outputs": [],
    "source": [
-    "from math import sqrt\n",
-    "from heapq import heapify\n",
+    "from math import sqrt, inf\n",
+    "from itertools import product\n",
     "\n",
-    "def euclidean_heuristic_cost(curr, end,grid):\n",
+    "def euclidean_heuristic_cost(curr, end):\n",
     "    \"\"\"\n",
     "    Estimates cost from curr (x0,y0) to end (x1,y1) using Euclidean\n",
     "    distance.\n",
     "    \"\"\"\n",
-    "    cost = grid[curr[0]][curr[1]] \n",
-    "    a = cost if cost != 0 else 0.001\n",
-    "\n",
-    "    ret = (a * sqrt((curr[0]-end[0])**2 + (curr[1]-end[1])**2))\n",
-    "    \n",
-    "    return ret\n",
+    "    currx, curry = curr\n",
+    "    endx, endy = end\n",
+    "    return sqrt((currx-endx)**2 + (curry-endy)**2)\n",
```

Comment:
What affect does this have on the performance of the astar algorithm? When I made a similar change it became unreliable.

Reply:
this probably won't make much of a difference. the plus side is that you do tuple assignment instead of indexing